### PR TITLE
Fix several coverity issues

### PR DESCRIPTION
**Pull request by nhorman**
coverity found some issues with the recently merged RCU implementation:
5191470
5191471
5191472
5191473
5191474
5191475
5191476
5191477

This PR contains fixes for all of them
There are two others  5191469 5191478 which are false positives and will require teaching coverity to accept them

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
CI is relevant.

**Review by shahsb — APPROVED** [review]

File: crypto/threads_win.c
Comment (t8m) [inline crypto/threads_win.c:127]: Nit: Missing empty line after declarations

File: crypto/threads_pthread.c
Comment (t8m) [inline crypto/threads_pthread.c:288]: Nit: missing empty line after declarations

**Review by t8m — CHANGES_REQUESTED** [review]

File: crypto/threads_win.c
Comment (nhorman) [reply crypto/threads_win.c:128]: corrected

File: crypto/threads_pthread.c
Comment (nhorman) [reply crypto/threads_pthread.c:288]: corrected

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
ping for second review

**Comment by nhorman** [comment]
ping, still need one more review here

**Comment by mattcaswell** [comment]
Looks like this needs a rebase

File: crypto/threads_pthread.c
Comment (mattcaswell) [inline crypto/threads_pthread.c:292]: This looks suspect the second argument in this call should be an OSSL_LIB_CTX.....there is no default libctx in the FIPS module.

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: crypto/threads_pthread.c
Comment (nhorman) [reply crypto/threads_pthread.c:292]: oh, you know what, with the recent context changes in rcu, this now conflicts and needs fixing up.  will do

Comment (nhorman) [reply crypto/threads_pthread.c]: rebased, and fixed

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master.

**Pull request closed without merging**
